framework/src/dirackernels/DiracKernelInfo.C

```diff
@@ -33,11 +33,27 @@ DiracKernelInfo::addPoint(const Elem * elem, Point p)
 {
   _elements.insert(elem);
 
-  if (!hasPoint(elem, p))
+
+  auto & multi_point_list = _points[elem];
+
+  const unsigned int npoint = multi_point_list.first.size();
+  mooseAssert(npoint == multi_point_list.second.size(), "Different sizes for location and multiplicity data");
+
+  for (unsigned int i = 0; i < npoint; ++i)
   {
-    std::vector<Point> & point_list = _points[elem];
-    point_list.push_back(p);
+    Real delta = (multi_point_list.first[i] - p).norm_sq();
+
+    if (delta < TOLERANCE*TOLERANCE)
```

Comment:
This `TOLERANCE*TOLERANCE` is in a couple of places... can you codify this into one function so that these don't get out of sync?